types/n3/index.d.ts

```diff
@@ -95,32 +95,36 @@ export class Quad extends BaseQuad implements RDF.Quad {
 
 export class Triple extends Quad implements RDF.Quad {}
 
-export namespace DataFactory {
-    function namedNode<Iri extends string = string>(value: Iri): NamedNode<Iri>;
-    function blankNode(value?: string): BlankNode;
-    function literal(value: string | number, languageOrDatatype?: string | RDF.NamedNode): Literal;
-    function variable(value: string): Variable;
-    function defaultGraph(): DefaultGraph;
-    function quad(
+export interface DataFactoryInterface<Q_In extends RDF.BaseQuad = RDF.Quad, Q_Out extends BaseQuad = Quad>
+    extends RDF.DataFactory<Q_In, Q_Out>
+{
+    namedNode<Iri extends string = string>(value: Iri): NamedNode<Iri>;
+    blankNode(value?: string): BlankNode;
+    literal(value: string | number, languageOrDatatype?: string | RDF.NamedNode): Literal;
+    variable(value: string): Variable;
+    defaultGraph(): DefaultGraph;
+    quad(
         subject: RDF.Quad_Subject,
         predicate: RDF.Quad_Predicate,
         object: RDF.Quad_Object,
         graph?: RDF.Quad_Graph,
     ): Quad;
-    function quad<Q_In extends RDF.BaseQuad = RDF.Quad, Q_Out extends BaseQuad = Quad>(
+    quad<Q_In extends RDF.BaseQuad = RDF.Quad, Q_Out extends BaseQuad = Quad>(
         subject: Q_In["subject"],
         predicate: Q_In["predicate"],
         object: Q_In["object"],
         graph?: Q_In["graph"],
     ): Q_Out;
-    function triple(subject: RDF.Quad_Subject, predicate: RDF.Quad_Predicate, object: RDF.Quad_Object): Quad;
-    function triple<Q_In extends RDF.BaseQuad = RDF.Quad, Q_Out extends BaseQuad = Quad>(
+    triple(subject: RDF.Quad_Subject, predicate: RDF.Quad_Predicate, object: RDF.Quad_Object): Quad;
+    triple<Q_In extends RDF.BaseQuad = RDF.Quad, Q_Out extends BaseQuad = Quad>(
         subject: Q_In["subject"],
         predicate: Q_In["predicate"],
         object: Q_In["object"],
     ): Q_Out;
 }
 
+export const DataFactory: DataFactoryInterface;
```

Comment:
I wonder if changing a namespace export into an interface + export const won't be an accidental breaking change. I recall looking at this very piece of n3 types but I ran into from issues IIRC

Reply:
I've run tests across a few of the libs that I commonly use and no issues; will also run on some bigger libs such as Comunica.

@tpluscode would you also be able to test this against any libs you have that you're concerned will break with this change?

Reply:
Bump @tpluscode - I've also called for people to run tests on codebases they are concerned about in https://github.com/rdfjs/N3.js/issues/450; but have no responses.

Could you run tests on any codebases you're concerned about for yourself and then approve this PR if there are no issues?

Reply:
Will see to it after the weekend

Reply:
Hey. I did not see any issues but it also seems that I did not find any usages of the data factory directly fro n3.

Reply:
Thanks for looking!